Controller/Admin/TransactionTab.php

```diff
@@ -73,19 +73,32 @@ protected function _getListDataFromArray(array $aArray, string $sTransactionStat
         $aListData = [];
 
         foreach ($aArray as $sKey => $sValue) {
-            // add current transaction state as a hint if it differs from the response transaction state
-            if ($sTransactionState && $sKey === 'transactionState') {
-                $sValue = $sTransactionState !== Transaction::STATE_AWAITING
-                    ? $sValue
-                    : $sValue . ' (confirmation awaiting)';
-            }
-
             $aListData[] = [
                 'title' => Helper::translate($sKey),
-                'value' => $sValue,
+                'value' => $this->_getTransactionStateText($sKey, $sValue, $sTransactionState),
             ];
         }
 
         return $aListData;
     }
+
+    /**
+     * Adds the current transaction state as a hint for the merchant if it differs from the response transaction state
+     *
+     * @param string $sKey
+     * @param string $sValue
+     * @param string $sTransactionState
+     *
+     * @return string the transaction state text
+     */
+    private function _getTransactionStateText($sKey, $sValue, $sTransactionState = null)
+    {
+        if ($sTransactionState && $sKey === 'transactionState') {
```

Comment:
move the string 'transactionState' to a constant

Reply:
👍